pkg/tipselect/urts.go

```diff
@@ -0,0 +1,485 @@
+package tipselect
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/iotaledger/hive.go/events"
+	"github.com/iotaledger/hive.go/syncutils"
+
+	"github.com/gohornet/hornet/pkg/dag"
+	"github.com/gohornet/hornet/pkg/model/hornet"
+	"github.com/gohornet/hornet/pkg/model/milestone"
+	"github.com/gohornet/hornet/pkg/model/tangle"
+	"github.com/gohornet/hornet/pkg/utils"
+)
+
+// Score defines the score of a tip.
+type Score int
+
+// TipSelectionFunc is a function which performs a tipselection and returns two tips.
+type TipSelectionFunc = func() (hornet.Hashes, error)
+
+// TipSelStats holds the stats for a tipselection run.
+type TipSelStats struct {
+	// The duration of the tip-selection for a single tip.
+	Duration time.Duration `json:"duration"`
+	// The amount of lazy tips found and removed during the tip-selection.
+	LazyTips int `json:"lazy_tips"`
+}
+
+// TipCaller is used to signal tip events.
+func TipCaller(handler interface{}, params ...interface{}) {
+	handler.(func(*Tip))(params[0].(*Tip))
+}
+
+// WalkerStatsCaller is used to signal tip selection events.
+func WalkerStatsCaller(handler interface{}, params ...interface{}) {
+	handler.(func(*TipSelStats))(params[0].(*TipSelStats))
+}
+
+const (
+	// ScoreLazy is a lazy zip and should not be selected.
+	ScoreLazy Score = iota
+	// ScoreSemiLazy is a somewhat lazy tip.
+	ScoreSemiLazy
```

Comment:
do you want to call is `ScoreSomewhatLazy`? It sounds weird, but uses the terminology from the RFC

Reply:
OK, the RFC mixes up definitions and call it semi lazy later